translator/internaldata/metrics_to_oc.go

```diff
@@ -0,0 +1,362 @@
+// Copyright 2019 OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package internaldata
+
+import (
+	"sort"
+
+	ocmetrics "github.com/census-instrumentation/opencensus-proto/gen-go/metrics/v1"
+	"github.com/golang/protobuf/ptypes/wrappers"
+
+	"github.com/open-telemetry/opentelemetry-collector/consumer/consumerdata"
+	"github.com/open-telemetry/opentelemetry-collector/internal"
+	"github.com/open-telemetry/opentelemetry-collector/internal/data"
+)
+
+const (
+	invalidMetricDescriptorType = ocmetrics.MetricDescriptor_Type(-1)
+)
+
+type labelKeys struct {
+	// ordered OC label keys
+	keys []*ocmetrics.LabelKey
+	// map from a label key literal
+	// to its index in the slice above
+	keyIndices map[string]int
+}
+
+func MetricDataToOC(td data.MetricData) []consumerdata.MetricsData {
+	ocMetricsData := consumerdata.MetricsData{}
+
+	resourceMetricsList := td.ResourceMetrics()
+
+	ocResourceMetricsList := make([]consumerdata.MetricsData, 0, len(resourceMetricsList))
+
+	for _, resourceMetrics := range resourceMetricsList {
+		ocMetricsData.Node, ocMetricsData.Resource = internalResourceToOC(resourceMetrics.Resource())
+		ocMetrics := make([]*ocmetrics.Metric, 0, resourceMetrics.MetricCount())
+		for _, instrumentationLibraryMetrics := range resourceMetrics.InstrumentationLibraryMetrics() {
+			// TODO: Handle instrumentation library name and version.
+			metrics := instrumentationLibraryMetrics.Metrics()
+			for _, metric := range metrics {
+				ocMetrics = append(ocMetrics, metricToOC(metric))
+			}
+		}
+		ocMetricsData.Metrics = ocMetrics
+		ocResourceMetricsList = append(ocResourceMetricsList, ocMetricsData)
+	}
+
+	return ocResourceMetricsList
+}
+
+func metricToOC(metric data.Metric) *ocmetrics.Metric {
+	labelKeys := labelKeysToOC(metric)
+	return &ocmetrics.Metric{
+		MetricDescriptor: descriptorToOC(metric.MetricDescriptor(), labelKeys),
+		Timeseries:       dataPointsToTimeseries(metric, labelKeys),
+		Resource:         nil,
+	}
+}
+
+func labelKeysToOC(metric data.Metric) *labelKeys {
```

Comment:
nit: by the name I was expecting something in OC format but it is actually something intermediary

Reply:
Agree, a bit confusing. Changed to `collectLabelKeys`